public/app/features/dashboard/components/TimePicker/TimePickerOptionGroup.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { PureComponent, createRef } from 'react';
+import { GroupProps } from 'react-select/lib/components/Group';
+
+interface Props extends GroupProps<any> {
+  data: {
+    label: string;
+    options: any[];
+    onCustomClick: (ref: any) => void;
+  };
+}
+
+export class TimePickerOptionGroup extends PureComponent<Props> {
+  pickerTriggerRef = createRef<HTMLDivElement>();
```

Comment:
I think it's this component responsibility to hide some of the custom popover logic that's in `TimePicker` now. It could render sth like `CustomTimePicker` component, that would render the custom time picker popover and it's trigger (label + icon in this case).